### DISCUSSION TITLE
Curiosity Nano - choose which to upload to

### DISCUSSION THREAD

**Discussion by MX682X** [SpenceKonde/DxCore#176] [Future development discussion]
Is it possible to choose the Curiosity Nano to which a sketch is supposed to be uploaded to from the Arduino IDE? From my experience it always uploads to the device with the smallest Serial number.

**Comment by SpenceKonde** [comment]
Nope. This is controlled by avrdude, I'd tell you to raise the issue with the maintainer, except one hasn't existed for like half a decade, so the only fixes we get are half-baked, half-assed and only if the consequences of the status quo are catastrophic (like when curiosity nano didn't work at all for Dx, there was a bunch of fingerpointing for a while, then someone with Microchip went in added support and crept out before anyone figured out that there was someone who knows how to fix problems in avrdude, and surrounded him waving bug reports and complaints. SerialUPDI will never be brought back in the direction of being able to do USB uploads, but a second fork of pcmcuprog aimed at that could be adapted to do it, except that there is no way to populate a menu that the target could be selected from in the IDE :-/

**Comment by MX682X** [comment]
I probably crazy, but I'm thinking about forking avrdude, adding a feature that checks if a com port is specified, if it is, checks if this is an curiosity nano through PID/VID, and then uses it's Serial number to find the right programmer. The DxCore, on the other hand, only adds the com port to the command if Curiosity Nano is selected as programmer....

Well, it's nothing I would do in the next cople of years for sure.... And it would probably only work on Windows....
One can dram, right?

**Comment by SpenceKonde** [comment]
I've got three fixed I could make to avenue in an instant to improve the UX
significantly. I've gone as far as to find the exact line of code where
they are.

But I don't know how to build that shit for 6 platforms, nor do I want to
become known as the guy who can fix avrdude.


(The three things, BTW: remove the unnecessary 256b cap on read size for
jtag2updi. Remove the spurious warning about boot memory on jtag2updi. And
fix the signature error message on classic avrs to tell people something
helpful instead of mentioning -f which will never ever fix the damned
problem (maybe show the sig that it read at the first level of verbosity
that displays an error at all, and if it read 0x000000, "this is either a
wiring problem, or the chip is set to use an external clock or crystal
which is not present or not working." 0xFFFFFF would tell them its a wiring
problem (that one always is) and the ID of a different part would say
something like this "the selected part has ID (xxxxxxx), probably a
(yyyyyy), but saw ID (xxxxxxx), probably a (yyyyyy). Check that the correct
port and board type is selected.

____________
Spence Konde
Azzy’S Electronics

New products! Check them out at tindie.com/stores/DrAzzy
GitHub: github.com/SpenceKonde
ATTinyCore: Arduino support for almost every ATTiny microcontroller
Contact: ***@***.***
